src/util/invariant.h

```diff
@@ -254,14 +269,18 @@ inline void invariant_violated_string(
 // This condition should be used to document that assumptions that are
 // made on goto_functions, goto_programs, exprts, etc. being well formed.
 // "The data structure is corrupt or malformed"
+//
+// It is okay for a DATA_INVARIANT to have an empty string as a reason string
+// if you can't state the reason, or doing so doesn't add any information -
+// just restates the condition.
```

Comment:
I disagree.  The reason DATA_INVARIANT has a condition is to help us document what we assume about key data structures, so that when we get to https://github.com/diffblue/cbmc/issues/751 we have an idea of what to put.  If in doubt, the comment should say which object you think this is a property of; for example

DATA_INVARIANT(it != goto_program.end(), "Target of goto must be in the same goto program.");

Reply:
I see your point, I will fix this. Thanks.